reproman/interface/tests/test_execute.py

```diff
@@ -143,3 +148,41 @@ def test_trace_local(trace_info):
 
     expect = {"packages": [{"files": ["/bin/ls"], "name": "coreutils"}]}
     assert_is_subset_recur(expect, attr.asdict(deb_dists[0]), [dict, list])
+
+
+@mark.skipif_no_network
+@mark.skipif_no_docker_engine
+def test_docker_shim():
+    shim = op.join(op.dirname(op.realpath(__file__)),
+            "../docker.shim")
+
+    image_digest = 'debian@sha256:a94839ed73dff831b2382b087b1008877d796946cc716afd4fc72150e082d1b8'
+    command = shim + " --debug run --rm {} cat /etc/debian_version".format(
+        image_digest)
+
+    tempdir = tempfile.mkdtemp()
+
+    env = {
+        "REPROMAN_TRACERS_DIR": tempdir,
+        "REPROMAN_EXTRA_TRACE_FILE": op.join(tempdir, "trace.extra.out")
+    }
+
+    # Test with theoretically successful docker run
+    with swallow_logs(new_level=logging.DEBUG) as log:
+        Runner().run(command, env=env)
+        assert_in('buster/sid', log.lines)
+        assert_in('[DEBUG DOCKER SHIM] Found docker executable: /usr/bin/docker', log.lines)
+        assert_in('[DEBUG DOCKER SHIM] Found digest ID: {}'.format(image_digest), log.lines)
+        assert op.exists(op.join(tempdir, env['REPROMAN_EXTRA_TRACE_FILE']))
```

Comment:
These assert_in (and assert_in_in) calls could just be `assert X in log.out`, which would could then take advantage of pytest's assertion introspection.